src/dialog/dialog.scss

```diff
@@ -0,0 +1,224 @@
+  @media screen and (max-width: 400px) {
+    max-width: 95%;
+    width: 95%;
+    min-width: 95%;
+  }
+
+  // Modal Animations for CSS
+  visibility: hidden;
+  opacity: 0;
+  transition: visibility $iui-speed-instant linear $iui-speed-fast, opacity $iui-speed-fast ease-out;
+
+  &.iui-dialog-visible {
+    visibility: visible;
+    opacity: 1;
+    // remove delay for entry animation so modal is instantly visible
+    transition-delay: $iui-speed-instant;
+  }
+}
+
+// @mixin iui-dialog-backdrop {
+//   @include iui-reset;
+//   z-index: 999;
+//   isolation: isolate;
+//   position: fixed;
+//   top: 0;
+//   left: 0;
+//   width: 100%;
+//   height: 100%;
+//   @include themed {
+//     background-color: rgba(0, 0, 0, t(iui-opacity-4));
+//   }
+
+//   // Modal Animations for CSS
+//   visibility: hidden;
+//   opacity: 0;
+//   transition: visibility $iui-speed-instant linear $iui-speed-fast, opacity $iui-speed-fast ease-out;
+
+//   &.iui-dialog-visible {
+//     visibility: visible;
+//     opacity: 1;
+//     // remove delay for entry animation so modal is instantly visible
+//     transition-delay: $iui-speed-instant;
+//   }
+// }
+
+// @mixin iui-dialog-modal { //iui-modal-default
+//   .iui-dialog {
+//     left: 50%;
+//     top: 33%;
+//     transform: translate(-50%, -33%);
+//     max-width: 50%;
+//     min-width: 380px;
+//     max-height: 100vh;
+
+//     @media screen and (max-width: 400px) {
+//       max-width: 95%;
+//       width: 95%;
+//       min-width: 95%;
+//     }
+//   }
+// }
+
+// @mixin iui-dialog-modal-full-page {
+//   $iui-full-page-modal-slide-in-speed: 0.3s;
+//   $iui-full-page-modal-slide-out-speed: 0.25s;
+
+//   @media (prefers-reduced-motion: no-preference) {
+//     transition: visibility $iui-speed-instant linear $iui-speed-slow, opacity $iui-speed ease-out $iui-speed-fast;
+//   }
+
+//   .iui-dialog {
+//     border-radius: 0;
+//     height: 100vh;
+//     width: 100vw;
+//     top: 0;
+//     left: 0;
+//     max-width: initial;
+//     min-width: initial;
+//     display: flex;
+//     flex-direction: column;
+//     will-change: transform;
+//     @media (prefers-reduced-motion: no-preference) {
+//       transform: translateY(100%);
+//       transition: visibility $iui-speed-instant linear $iui-speed, opacity $iui-speed-instant linear $iui-speed,
+//         transform $iui-full-page-modal-slide-out-speed ease-in;
+//     }
+//   }
+
+//   &.iui-dialog-visible .iui-dialog {
+//     transform: translateY(0);
+//     @media (prefers-reduced-motion: no-preference) {
+//       transition: transform $iui-full-page-modal-slide-in-speed ease-out;
+//     }
+//   }
+// }
+
+@mixin iui-dialog-draggable {
+  display: flex;
+  flex-direction: column;
+  @include themed {
+    border: 1px solid t(iui-color-background-border);
+  }
+}
+
+@mixin iui-dialog-resizable {
+  max-width: 100vw;
+  max-height: 100vh;
+  min-width: 380px;
+  min-height: 144px;
+  resize: both;
+}
+
+@mixin iui-dialog-animation {
```

Comment:
Why separate mixin and class for animation? Just have it on the base dialog.

Reply:
In case you wanted to re-use this mixin to get the modal styling without the animation?  I can't think of any other benefit.

Reply:
But do we want to do that?